src/plugins/navigation/public/top_nav_menu/top_nav_menu.test.tsx

```diff
@@ -133,6 +136,7 @@ describe('TopNavMenu', () => {
             fullWidth: true,
             activeOption: [{ label: 'what', id: '1' }],
           },
+          dataSourceSelection,
```

Comment:
Does it mean all the places that are using data source pickers need to pass the dataSourceSelection explicitly?  I thought it is an internal state and plugins do not need to be aware of this context.

Reply:
There is no need to pass this instance manually, because usually all components are created by functions provided by DSM and passed within the function.  Except for some test cases when manually rendering components that need to pass.